src/core/Action.cpp

```diff
@@ -210,21 +220,56 @@ void Action::checkRead() {
     }
     error(msg);
   }
+  setupConstantValues(false);
+}
+
+void Action::setupConstantValues( const bool& have_atoms ) {
+  if( have_atoms ) {
+    // This ensures that we switch off actions that only depend on constant when passed from the
+    // MD code on the first step
+    ActionAtomistic* at = dynamic_cast<ActionAtomistic*>( this );
+    ActionWithValue* av = dynamic_cast<ActionWithValue*>( this );
+    if( at && av ) {
+      never_activate=av->getNumberOfComponents()>0;
+      for(unsigned i=0; i<av->getNumberOfComponents(); ++i) {
+        if( !av->copyOutput(i)->isConstant() ) { never_activate=false; break; }
+      }
+    }
+  }
+  ActionWithArguments* aa = dynamic_cast<ActionWithArguments*>( this );
+  if(aa) never_activate = aa->calculateConstantValues( have_atoms );
 }
 
 long long int Action::getStep()const {
   return plumed.getStep();
 }
 
 double Action::getTime()const {
-  return plumed.getAtoms().getTimeStep()*getStep();
+  return timestep*getStep();
 }
 
 double Action::getTimeStep()const {
-  return plumed.getAtoms().getTimeStep();
+  return timestep;
 }
 
-
+double Action::getkBT() {
+  double temp=-1.0;
+  if( keywords.exists("TEMP") ) parse("TEMP",temp);
+  if(temp>=0.0 && keywords.style("TEMP","optional") ) return getKBoltzmann()*temp;
```

Comment:
@gtribello we could keep the consistency check with the MD temperature also when TEMP is optional, no?

Reply:
@invemichele I didn't do it in the way you suggest because I wanted to maintain backwards compatibility.  When people have specified the TEMP keyword as optional there is no warning. Compare what is done in the master version of METAD:

```c++
parse("TEMP",temp);
if(temp>0.0) kbt_=plumed.getAtoms().getKBoltzmann()*temp;
else kbt_=plumed.getAtoms().getKbT();
```

with the ExpansionCVs code:

```c++
parse("TEMP",temp);
if(temp>0)
  {
    if(kbt_>0 && std::abs(kbt_-kB*temp)>1e-4)
      log.printf(" +++ WARNING +++ using TEMP=%g while MD engine uses %g\n",temp,kbt_/kB);
    kbt_=kB*temp;
  }
  plumed_massert(kbt_>0,"your MD engine does not pass the temperature to plumed, you must specify it using TEMP");
  log.printf("  temperature = %g, beta = %g\n",kbt_/kB,1./kbt_);
``` 

The code above should produce the same result in these two cases.